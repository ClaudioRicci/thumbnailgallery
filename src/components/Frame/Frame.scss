.card .frame img {
  width: 100%;
  height: auto;
  margin: 0;
}
