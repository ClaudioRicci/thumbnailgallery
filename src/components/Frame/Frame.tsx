import React, { useState, memo } from "react";
import Modal from "../Modal/Modal";
import "./Frame.scss";

function Frame(props: any) {
  const [modalOpen, setModalOpen] = useState(false);
  return (
    <article key={props.id} className="frame" data-testid="frame">
      <img
        src={props.large_url}
        alt={
          "Image is copyright of " +
          props.copyright +
          ", sourced by " +
          props.site +
          "."
        }
      />
      {modalOpen && <Modal {...props} onClose={() => setModalOpen(false)} />}
    </article>
  );
}

Frame.defaultProps = {
  id: 0,
  large_url: "https://splashbase.s3.amazonaws.com/unsplash/large/14ZGHd9",
  copyright: "CCO",
  site: "unknown"
};

export default memo(Frame);
