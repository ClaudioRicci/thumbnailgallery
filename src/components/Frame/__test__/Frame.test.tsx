import React from "react";
import ReactDOM from "react-dom";
import Frame from "../Frame";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("Renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Frame></Frame>, div);
});

test("Renders Frame Correctly", done => {
  const { getByTestId } = render(<Frame />);
  expect(getByTestId("frame")).toHaveTextContent("");
  done();
});

it("Matches snapshot", () => {
  const tree = renderer.create(<Frame></Frame>).toJSON();
  expect(tree).toMatchSnapshot();
});
