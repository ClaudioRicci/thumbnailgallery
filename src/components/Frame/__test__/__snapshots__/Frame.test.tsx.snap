// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<article
  className="frame"
  data-testid="frame"
>
  <img
    alt="Image is copyright of CCO, sourced by unknown."
    src="https://splashbase.s3.amazonaws.com/unsplash/large/14ZGHd9"
  />
</article>
`;
